core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/catalog/CamelModelFactory.java

```diff
@@ -181,6 +181,10 @@ public static String getCompatibleCamelVersion(String requestedCamelVersion) {
 	 */
 	public static String getLatestCamelVersion() {
 		String latest = null;
+		// make sure we're initialized here
+		if (supportedCamelModels == null || supportedCamelModels.keySet() == null) {
+		    initializeModels();
```

Comment:
if called from several places, I think that we need to create a singleton and have a synchronized block